SpatialGDK/Source/SpatialGDKEditorCommandlet/Private/Commandlets/CookAndGenerateSchemaCommandlet.cpp

```diff
@@ -0,0 +1,135 @@
+					*Object->GetFName().ToString(), *GetPathNameSafe(Object->GetClass()));
+				VisitedClasses->Add(SoftClass);
+			}
+			else
+			{
+				UnsupportedClasses.Add(SoftClass);
+			}
+		}
+	}
+
+private:
+
+	TSet<FSoftClassPath>* VisitedClasses;
+	TSet<FSoftClassPath> UnsupportedClasses;
+};
+
+UCookAndGenerateSchemaCommandlet::UCookAndGenerateSchemaCommandlet()
+{
+	IsClient = false;
+	IsEditor = true;
+	IsServer = false;
+	LogToConsole = true;
+}
+
+int32 UCookAndGenerateSchemaCommandlet::Main(const FString& CmdLineParams)
+{
+	UE_LOG(LogCookAndGenerateSchemaCommandlet, Display, TEXT("Cook and Generate Schema Started."));
+	FObjectListener ObjectListener;
+	TSet<FSoftClassPath> ReferencedClasses;
+	ObjectListener.StartListening(&ReferencedClasses);
+	
+	UE_LOG(LogCookAndGenerateSchemaCommandlet, Display, TEXT("Try Load Schema Database."));
+	if (!TryLoadExistingSchemaDatabase())
+	{
+		UE_LOG(LogCookAndGenerateSchemaCommandlet, Error, TEXT("Failed to load Schema Database."));
+		return 0;
+	}
+
+	UE_LOG(LogCookAndGenerateSchemaCommandlet, Display, TEXT("Finding supported C++ and in-memory Classes."));
+	for (const auto& SupportedClass : GetAllSupportedClasses())
+	{
+		ReferencedClasses.Add(FSoftClassPath(SupportedClass));
+	}
+
+	UE_LOG(LogCookAndGenerateSchemaCommandlet, Display, TEXT("Starting Cook Command."));
+	int32 CookResult = Super::Main(CmdLineParams);
+	UE_LOG(LogCookAndGenerateSchemaCommandlet, Display, TEXT("Cook Command Completed."));
+
+	UE_LOG(LogCookAndGenerateSchemaCommandlet, Display, TEXT("Discovered %d Classes during cook."), ReferencedClasses.Num());
+
+	ObjectListener.StopListening();
+
+	// Sort classes here so that batching does not have an effect on ordering.
+	ReferencedClasses.Sort([](const FSoftClassPath& A, const FSoftClassPath& B)
+	{
+		return A.GetAssetPathName() < B.GetAssetPathName();
+	});
+
+	UE_LOG(LogCookAndGenerateSchemaCommandlet, Display, TEXT("Start Schema Generation for discovered assets."));
+	FDateTime StartTime = FDateTime::Now();
+	TSet<UClass*> Classes;
+	const int BatchSize = 100;
+	for (const FSoftClassPath& SoftPath : ReferencedClasses)
+	{
+		if (UClass* LoadedClass = SoftPath.TryLoadClass<UObject>())
+		{
+			UE_LOG(LogCookAndGenerateSchemaCommandlet, Verbose, TEXT("Reloaded %s, adding to batch"), *GetPathNameSafe(LoadedClass));
+			Classes.Add(LoadedClass);
+			if (Classes.Num() >= BatchSize)
+			{
+				SpatialGDKGenerateSchemaForClasses(Classes);
+				Classes.Empty(BatchSize);
+			}
+		}
+		else
+		{
+			UE_LOG(LogCookAndGenerateSchemaCommandlet, Warning, TEXT("Failed to reload %s"), *SoftPath.ToString());
+		}
+	}
+	SpatialGDKGenerateSchemaForClasses(Classes);
+
+	FTimespan Duration = FDateTime::Now() - StartTime;
+
+	UE_LOG(LogCookAndGenerateSchemaCommandlet, Display, TEXT("Schema Generation Finished in %.2f seconds"), Duration.GetTotalSeconds());
+	
+	if (!SaveSchemaDatabase())
```

Comment:
Do we have a ticket to iterate on this process so we don't have to run the cook command multiple times?